health.go

```diff
@@ -70,14 +72,25 @@ type (
 )
 
 // Register registers a check config to be performed.
-func Register(c Config) {
+func Register(c Config) error {
 	mu.Lock()
 	defer mu.Unlock()
 
 	if c.Timeout == 0 {
 		c.Timeout = time.Second * 2
 	}
-	checks = append(checks, c)
+
+	if c.Name == "" {
+		return errors.New("health check must have a name to be registered")
+	}
+
```

Comment:
let's move the lock to here so in case of name is empty then we don't acquire one

Reply:
I see your point, however the Name checking might return an error, it's does not make sense to have the lock on top of the function anymore in case name is empty, IMO it's fair enough to move the lock down

Reply:
Sure, it is now moved.